site/docs/v1/tech/lambdas/testing-lambdas.md

```diff
@@ -0,0 +1,502 @@
+  } catch (e) {
+    console.error('Error creating user: ');
+    console.dir(e, { depth: null });
+    process.exit(1);
+  }
+}
+```
+
+The code above has three sections:
+
+- A declaration of constant variables.
+- Creation of the User request object with `const request`. Details on this can be found in the [TypeScript client library interface](https://github.com/FusionAuth/fusionauth-typescript-client/blob/master/src/FusionAuthClient.ts).
+- Making the request with the client library and checking the response with `await fusion.register(userId, request);`.
+
+Change the <span>applicationId</span>{:.filed} value at the top to match the application Id of your Application in FusionAuth and run the code with the following command.
+
+```bash
+node test.js
+```
+
+In FusionAuth, click on <span>Users</span>{:.breadcrumb} to check that a new user called `lambdatestuser` has been created. Now that you have a user profile to work with, you can delete the `createUser();` line in `test.js`.
+
+#### Write the Test
+
+Now you can use the new test user to test that the lambda returns "Goodbye World", which will confirm that the CLI `update` command worked.
+
+Add this code to the `test.js` file.
+
+```js
+test('test login returns JWT with "Goodbye World"', async function (t) {
+  t.plan(1);
+  const result = await login();
+  t.ok(result.toLowerCase().includes('goodbye world'));
+  t.end();
+});
+
+async function login() {
+  try {
+    const request  = {
+      applicationId: applicationId,
+      loginId: userEmail,
+      password: userPassword,
+    };
+    const fusion = new client.FusionAuthClient('lambda_testing_key', fusionUrl);
+    const clientResponse = await fusion.login(request);
+    if (!clientResponse.wasSuccessful)
+      throw Error(clientResponse);
+    const jwtToken = clientResponse.response.token;
+    const decodedToken = jwt.decode(jwtToken);
+    const message = decodedToken.message;
+    return message;
+  }
+  catch (e) {
+    console.error('Error: ');
+    console.dir(e, { depth: null });
+    process.exit(1);
+  }
+}
+```
+
+The `login` function calls the FusionAuth library in a similar way to `createUser`. It then decodes the JWT response and returns its `message` property.
+
+This `login` function is called by the `tape` function `test`. This test gives the name of the test, says that it expects exactly one assertion to occur with `plan`, checks that calling `login` returns the property you expect from our lambda updated earlier, and exits.
+
+Run it with the following command.
+
+```bash
+node test.js
+```
+
+When your code has several tests and you want a colorful concise summary, you can instead use the following.
+
+```bash
+node test.js | npx faucet
+```
+
+In reality, you'll want to create a random user at the beginning of each login test and remove it at the end. This will allow you to add multiple lambda tests while avoiding potential conflicts between test users and permissions.
+
```

Comment:
Would be great to have another unit test example where you are pulling attributes from the user object, performing logic on it, and then setting a value on a jwt. Maybe the user has a certain role ('admin', 'editor', 'viewer') and the JWT is given certain permissions based on those: (admin -> 'all', 'editor' -> 'read', 'write', viewer' -> 'read'). or something similar. But I think this kind of logic is where unit tests shine, plus showing how to mock up a user object would be useful.